File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        stack = []

        for token in tokens:
            if token == "+":
                x = stack.pop() + stack.pop()
                stack.append(x)
            elif token == "-":
                a = stack.pop()
                b = stack.pop()
                x = b - a
                stack.append(x)

            elif token == "*":
                x = stack.pop() * stack.pop()
                stack.append(x)

            elif token == "/":
                a = stack.pop()
                b = stack.pop()
                x = int(b / a) if b * a >= 0 else int(b / a) + 0
                stack.append(x)

            else:
                stack.append(int(token))
        
        
        return stack.pop()

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.java
class Solution {
    public int evalRPN(String[] tokens) {
        Stack<Integer>st=new Stack<>();
        for(int i=0;i<tokens.length;i++){
            if(tokens[i].equals("+")){
                int x=st.pop()+st.pop();
                st.push(x);
            }
            else if(tokens[i].equals("-")){
                int a=st.pop();
                int b=st.pop();
                int x=b-a;
                st.push(x);
            }
            else if(tokens[i].equals("*")){
                int x=st.pop()*st.pop();
                st.push(x);
            }
            else if(tokens[i].equals("/")){
                int a=st.pop();
                int b=st.pop();
                int x=b/a;
                st.push(x);
            }
            else{
                st.push(Integer.parseInt(tokens[i]));
            }
            
        }
        return st.pop();
    }
}